packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -149,20 +140,60 @@ const RelationModalBody = ({ id }: RelationModalBodyProps) => {
 
   const documentTitle = documentResponse.getTitle(documentLayoutResponse.edit.settings.mainField);
 
+  const getFullPageLink = (): string => {
+    const isSingleType = documentMeta.collectionType === SINGLE_TYPES;
+    const queryParams = documentResponse?.document?.locale
+      ? `?plugins[i18n][locale]=${documentResponse.document.locale}`
+      : '';
+
+    return `/content-manager/${documentMeta.collectionType}/${documentMeta.model}${isSingleType ? '' : '/' + documentMeta.documentId}${queryParams}`;
+  };
+
+  const editViewUrl = `${pathname}${search}`;
+  const modalDocumentUrlEqualEditViewUrl = editViewUrl.includes(getFullPageLink());
+
   const hasDraftAndPublished = documentResponse.schema?.options?.draftAndPublish ?? false;
 
   return (
     <Modal.Body>
       <DocumentRBAC permissions={permissions} model={documentMeta.model}>
-        <Flex direction="column" alignItems="flex-start" gap={2}>
-          <Typography tag="h2" variant="alpha">
-            {documentTitle}
-          </Typography>
-          {hasDraftAndPublished ? (
-            <Box marginTop={1}>
-              <DocumentStatus status={documentResponse.document?.status} />
-            </Box>
-          ) : null}
+        <Flex alignItems="flex-start" justifyContent="space-between" width="100%">
+          <Flex direction="column" alignItems="flex-start" gap={2}>
+            <Typography tag="h2" variant="alpha">
+              {documentTitle}
+            </Typography>
+            {hasDraftAndPublished ? (
+              <Box marginTop={1}>
+                <DocumentStatus status={documentResponse.document?.status} />
+              </Box>
+            ) : null}
+          </Flex>
+          <Flex>
+            {modalDocumentUrlEqualEditViewUrl ? (
+              <IconButton
+                onClick={onToggle}
+                variant="tertiary"
+                label={formatMessage({
+                  id: 'content-manager.components.RelationInputModal.button-fullpage',
+                  defaultMessage: 'Go to entry',
+                })}
+              >
+                <ArrowsOut />
+              </IconButton>
+            ) : (
+              <IconButton
+                tag={Link}
+                to={getFullPageLink()}
+                variant="tertiary"
+                label={formatMessage({
+                  id: 'content-manager.components.RelationInputModal.button-fullpage',
+                  defaultMessage: 'Go to entry',
+                })}
+              >
+                <ArrowsOut />
+              </IconButton>
+            )}
```

Comment:
Instead of the condition I think we could just always toggle the modal onClick. So if the rootDocument is both on the editview and the modal then the modal closes and we see it on the edit view, otherwise it closes the modal and navigates to the document.

Reply:
ok I need also to implement the Confirmation modal when we try to leave a modal and the modifed value of useForm is true (I need to implement the same in the case we navigate nested relations with field modified)